docs/abi.md

```diff
@@ -89,16 +89,14 @@ functions** as
 ### wait_on_channels
 
 `wait_on_channels(usize, u32) -> u32` blocks until data is available for reading
-from one of the specified channel handles. The channel handles are encoded in a
-buffer that holds N contiguous 9-byte chunks, each of which is made up of an
-8-byte channel handle value (little-endian `u64`) followed by a single channel
-status byte. Invalid handles will have an `INVALID_CHANNEL` status, but
-`wait_on_channels` return value will only fail for internal errors or if _all_
-channels are invalid.
-
-If reading from any of the specified (valid) channels would violate
-[information flow control](/docs/concepts.md#labels), returns
-`ERR_PERMISSION_DENIED`.
+from one of the specified channel handles, unless any of the channels is
+invalid, orphaned, or violates the
+[information flow control](/docs/concepts.md#labels). The channel handles are
+encoded in a buffer that holds N contiguous 9-byte chunks, each of which is made
+up of an 8-byte channel handle value (little-endian u64) followed by a single
+channel status byte. Invalid handles will have an `INVALID_CHANNEL`, `ORPHANED`,
+or `PERMISSION_DENIED` status, but `wait_on_channels` return value will only
+fail for internal errors or if _all_ channels are invalid.
```

Comment:
Is this text still correct?

Reply:
I think so, since this is about the `ABI`. But it is not the case for `wait_on_channels` in the `Runtime`. There is an extra check in the implementation of `wait_on_channels` in the `wasm` node that returns `Err(OakStatus::ErrBadHandle)` if all channels are invalid (perhaps I could update it to check for `ChannelReadStatus::PermissionDenied` too). 
```
        if statuses
            .iter()
            .all(|&s| s == ChannelReadStatus::InvalidChannel || s == ChannelReadStatus::Orphaned)
        {
            Err(OakStatus::ErrBadHandle)
        } else {
            Ok(())
        }
```

In this case the return value is just an extra flag. We already return all the updated statuses in the `status_buff` (arg 0).

Reply:
Ah yes, I misread it &ndash; it's describing the return value, not the conditions under which the call blocks (which is at the start).